neo/Consensus/ConsensusContext.cs

```diff
@@ -210,14 +210,45 @@ public ConsensusPayload MakePrepareRequest()
         {
             byte[] buffer = new byte[sizeof(ulong)];
             random.NextBytes(buffer);
+            Block.ConsensusData.Nonce = BitConverter.ToUInt64(buffer, 0);
+
             IEnumerable<Transaction> memoryPoolTransactions = Blockchain.Singleton.MemPool.GetSortedVerifiedTransactions();
             foreach (IPolicyPlugin plugin in Plugin.Policies)
                 memoryPoolTransactions = plugin.FilterForBlock(memoryPoolTransactions);
+
             List<Transaction> transactions = memoryPoolTransactions.ToList();
-            TransactionHashes = transactions.Select(p => p.Hash).ToArray();
-            Transactions = transactions.ToDictionary(p => p.Hash);
+
+            uint maxBlockSize;
+            Transactions = new Dictionary<UInt256, Transaction>();
+            using (var snapshot = Blockchain.Singleton.GetSnapshot())
```

Comment:
https://github.com/neo-project/neo/blob/34f37b9f93ee97df95a7d9583955dc6837826463/neo/Consensus/ConsensusContext.cs#L40

Should use `ConsensusContext.Snapshot`.